src/sources/kafka.rs

```diff
@@ -312,14 +311,19 @@ impl<'a> From<BorrowedMessage<'a>> for FinalizerEntry {
 
 fn mark_done(consumer: Arc<StreamConsumer<KafkaStatisticsContext>>) -> impl Fn(FinalizerEntry) {
     move |entry| {
-        if let Err(error) = consumer.store_offset(&entry.topic, entry.partition, entry.offset) {
+        let mut tpl = TopicPartitionList::new();
+        tpl.add_partition(&entry.topic, entry.partition)
+            .set_offset(Offset::from_raw(entry.offset + 1)) // Not sure why this needs a +1
+            .expect("Setting topic offset failed");
+        if let Err(error) = consumer.commit(&tpl, CommitMode::Sync) {
```

Comment:
Can you explain this change? I'm having a bit of a hard time groking how it differs. Also, should we not do this at all if auto commit is enabled?

Reply:
Before it was just storing the offset. This actually commits the offset, which is necessary for tests to read it back and for other consumers to pick up on. If we set `enable.auto.commit = false` then we have to commit offsets manually, which is what we want for acknowledgements. This function is only called by the finalizer callback, which is not used if acknowledgements are disabled.

Reply:
Thanks for explaining!

The `CommitMode::Sync` jumped out to me as potentially expensive; unfortunately we don't have any Kafka soaks to tell us. Maybe it'd make sense to run a couple of adhoc benchmarks to verify?

I found this FAQ about committing: I also found a FAQ about committing in https://github.com/edenhill/librdkafka/wiki/FAQ#why-committing-each-message-is-slow 

It mentions a few potential pitfalls like queues building up if committing is slower than processing. One alternative it suggests is relying on the autocommitter to commit on an interval and pushing offsets when you want to "ack". I'm curious what you think of that? That actually seems like it may have been the old behavior?

Reply:
I can see why you'd want to control when things are actually committed for tests though. We could maybe introduce some test-only code that modifies the behavior when testing multi-consumer behavior?

Reply:
Yeah, I see the problem. I had been thinking that these all happen in the background, but that would still serialize on the same connection. Yes, what you describe is indeed the old behavior. I'll see what I can do about test-only code. My concern would be testing ending up running different code than non-test runs.

Reply:
Yeah, that is definitely a concern. Maybe the tests could lower the autocommit interval to something like 100ms and then sleep in the tests?

Reply:
I have set the (auto)commit interval to 1ms, and still the committed offset is not updated after several seconds sleeping after all events are fetched when I only store the offset. 🤷🏻

Reply:
🤔 just to confirm, you also set `enable.auto.commit` back to `true`?

Reply:
No, that's the whole point of this patch, to stop the automatic storing of the offset when reading. I'm going to try to add some tests to validate this behavior though.

Reply:
I feel like we might by missing each other here 😄 

By my reading, `enable.auto.commit` doesn't, by itself, commit offsets when reading. If `enable.auto.offset.store` is `false`, as it is here, then the autocommit will only autocommit offsets on the interval once the offset is explicitly stored by `store_offset`.

My suggestion is to go back to the original behavior of auto-commit, but manual store of offsets. This should avoid the pitfall of slow commits causing queuing by letting the client still handle the commits on an interval. The trade-off is an increased risk of duplicate processing if Vector crashed, but I think that risk is acceptable.